src/widget/js/component/dialog.js

```diff
@@ -134,36 +139,63 @@ export default class Dialog {
     });
     this.file.on('change', () => {
       if (this.editable) {
-        let selectedFile = this.file.files[0];
-        this.ticket.channel.sendFileMessage(
-          selectedFile,
-          selectedFile.name,
-          selectedFile.type,
-          selectedFile.size,
-          '',
-          '',
-          [{ maxWidth: 220, maxHeight: 220 }],
-          (res, err) => {
-            if (!err) {
-              if (MessageElement.isVisible(res)) {
-                this.appendMessage(res);
-              }
+        const selectedFile = this.file.files[0];
+        const fileMessage = {
+          file: selectedFile,
+          fileName: selectedFile.name,
+          fileSize: selectedFile.size,
+          mimeType: selectedFile.type,
+          thumbnailSizes: [{ maxWidth: 220, maxHeight: 220 }],
+        };
+        this.ticket.channel.sendFileMessage(fileMessage)
+          .onPending((message) => {
+            // show progress
+          })
+          .onFailed((error, message) => {
+            // hide progress
+            console.log(error);
+          })
+          .onSucceeded((res) => {
+            if (MessageElement.isVisible(res)) {
+              // hide progress
+              this.appendMessage(res);
+              this.scrollToBottom();
             }
-          }
-        );
+          });
       }
     });
     this.file.on('click', e => e.stopPropagation());
   }
   loadMessage(next, callback) {
     if (!this.query || !next) {
-      this.query = this.ticket.channel.createPreviousMessageListQuery();
+      const filter = new MessageFilter();
+      const limit = MESSAGE_LIMIT;
+      const startingPoint = Date.now();
+      const collection = this.ticket.channel.createMessageCollection({
+        filter,
+        limit,
+        startingPoint,
+      });
```

Comment:
It seems doesn't need to define variables
```suggestion
      const collection = this.ticket.channel.createMessageCollection({
        filter: new MessageFilter(),
        limit: MESSAGE_LIMIT,
        startingPoint: Date.now(),
      });
```